api/desecapi/permissions.py

```diff
@@ -9,3 +9,12 @@ class IsOwner(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         return obj.owner == request.user
 
+
+class IsDomainOwner(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of a domain to view or edit an object owned by that domain.
```

Comment:
Misleading doc: this permission doesn't do anything on it's own; only if used it makes sure that the requests user is the object's domain owner.